File: rb/features.rb
# Cheapshark SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/ratelimit_feature'
require_relative 'feature/retry_feature'
require_relative 'feature/test_feature'
require_relative 'feature/timeout_feature'


module CheapsharkFeatures
  def self.make_feature(name)
    case name
    when "base"
      CheapsharkBaseFeature.new
    when "ratelimit"
      CheapsharkRatelimitFeature.new
    when "retry"
      CheapsharkRetryFeature.new
    when "test"
      CheapsharkTestFeature.new
    when "timeout"
      CheapsharkTimeoutFeature.new
    else
      CheapsharkBaseFeature.new
    end
  end
end

File: py/features.py
# Cheapshark SDK feature factory

from feature.base_feature import CheapsharkBaseFeature
from feature.test_feature import CheapsharkTestFeature


def _make_feature(name):
    features = {
        "base": lambda: CheapsharkBaseFeature(),
        "test": lambda: CheapsharkTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
